tests/lib.rs

```diff
@@ -478,116 +494,46 @@ fn parsing_binary_data_should_fail_cleanly() {
-                    }),
-                },
-                Part {
-                    uri: "part1.ts".into(),
-                    duration: 0.5,
-                    independent: false,
-                    gap: false,
-                    byte_range: Some(ByteRange {
-                        length: 50000,
-                        offset: Some(50000),
-                    }),
-                },
-            ],
-            ..Default::default()
-        }],
-        unknown_tags: vec![],
-        server_control: Some(ServerControl {
-            can_skip_until: Some(12.0),
-            can_skip_dateranges: false,
-            hold_back: Some(3.0),
-            part_hold_back: Some(1.5),
-            can_block_reload: true,
-        }),
-        part_inf: Some(PartInf { part_target: 0.5 }),
-        skip: Some(Skip {
-            skipped_segments: 3,
-        }),
-        preload_hint: Some(PreloadHint {
-            hint_type: "PART".into(),
-            uri: "next_part.ts".into(),
-            byte_range: Some(ByteRange {
-                length: 50000,
-                offset: Some(100000),
-            }),
-        }),
-        rendition_report: Some(RenditionReport {
-            uri: "rendition.m3u8".into(),
-            last_msn: Some(338559),
-            last_part: Some(1),
-        }),
-    });
-    let playlist_parsed = print_create_and_parse_playlist(&mut playlist_original);
-    assert_eq!(playlist_original, playlist_parsed);
+    let playlist_original = "\
+#EXTM3U
+#EXT-X-VERSION:6
+#EXT-X-INDEPENDENT-SEGMENTS
+#EXT-X-SERVER-CONTROL:CAN-SKIP-UNTIL=10,PART-HOLD-BACK=1.533,CAN-BLOCK-RELOAD=YES
+#EXT-X-PART-INF:PART-TARGET=0.511
+#EXT-X-TARGETDURATION:1
+#EXT-X-MEDIA-SEQUENCE:1
+#EXT-X-MAP:URI=\"init_track1504_.mp4\"
+#EXT-X-PROGRAM-DATE-TIME:2024-12-17T16:10:10.190Z
+#EXTINF:0.998467,
+1_track1504_.m4s
+#EXTINF:0.998456,
+2_track1504_.m4s
+#EXTINF:0.998456,
+3_track1504_.m4s
+#EXTINF:0.998456,
+4_track1504_.m4s
+#EXTINF:0.998467,
+5_track1504_.m4s
+#EXTINF:0.998456,
+6_track1504_.m4s
+#EXTINF:0.998456,
+7_track1504_.m4s
+#EXT-X-PART:URI=\"8_track1504_.m4s.1\",DURATION=0.510644,INDEPENDENT=YES
+#EXT-X-PART:URI=\"8_track1504_.m4s.2\",DURATION=0.487433,INDEPENDENT=YES
+#EXTINF:0.998456,
+8_track1504_.m4s
+#EXT-X-PART:URI=\"9_track1504_.m4s.1\",DURATION=0.510644,INDEPENDENT=YES
+#EXT-X-PART:URI=\"9_track1504_.m4s.2\",DURATION=0.487433,INDEPENDENT=YES
+#EXTINF:0.998456,
+9_track1504_.m4s
+#EXT-X-PART:URI=\"10_track1504_.m4s.1\",DURATION=0.510644,INDEPENDENT=YES
+#EXT-X-PART:URI=\"10_track1504_.m4s.2\",DURATION=0.487433,INDEPENDENT=YES
+#EXTINF:0.998467,
+10_track1504_.m4s
+#EXT-X-PART:URI=\"11_track1504_.m4s.1\",DURATION=0.510644,INDEPENDENT=YES
+#EXT-X-PRELOAD-HINT:TYPE=PART,URI=\"11_track1504_.m4s.2\"
+#EXT-X-RENDITION-REPORT:URI=\"playlist_1.m3u8\",LAST-MSN=1,LAST-PART=8
+";
+    let playlist_parsed = print_parse_and_create_playlist(&playlist_original);
+    assert_eq!(playlist_parsed, playlist_original);
```

Comment:
You should add a test that builds the playlist from code

Reply:
Why? the current test deserialize the String to Objects, and Objects to String, my feeling is that the test scope takes everything we need, from parsing to serializing

Reply:
Ok, but that test serves as an example of how to create use the API, and you verify that the API is accessible and works.

Reply:
Done